src/tools/rbd_mirror/image_map/StateTransition.h

```diff
@@ -0,0 +1,59 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_RBD_MIRROR_IMAGE_MAP_STATE_TRANSITION_H
+#define CEPH_RBD_MIRROR_IMAGE_MAP_STATE_TRANSITION_H
+
+namespace rbd {
+namespace mirror {
+namespace image_map {
+
+class StateTransition {
+public:
+  enum ActionType {
+    ACTION_TYPE_ADD = 0,
+    ACTION_TYPE_REMOVE,
+    ACTION_TYPE_SHUFFLE,
+  };
+
+  enum SubActionType {
+    SUB_ACTION_TYPE_MAP = 0,
+    SUB_ACTION_TYPE_UNMAP,
+    SUB_ACTION_TYPE_PURGE,
```

Comment:
not sure I understood this fully -- would this mean having on-disk map updates and acquiring/removing images as two separate passes (two state-machine transitions)?

Reply:
Yeah -- personally I think it would be clearer in that it shows the true state of the system (waiting on an on-disk update vs waiting for a instance to ACK the acquire/release). Perhaps it make sense to keep one as UPDATING_MAPPING and another as REMOVING_MAPPING if that helps get objects in the correct collection.